Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -1107,7 +1106,12 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
         // all the fields when sending the response to the WebContent process over IPC.
         resourceResponse.disableLazyInitialization();
 
-        resourceResponse.setDeprecatedNetworkLoadMetrics(WebCore::copyTimingData(taskMetrics, networkDataTask->networkLoadMetrics()));
+        auto responseMetrics = WebCore::copyTimingData(taskMetrics, networkDataTask->networkLoadMetrics());
+        if (responseMetrics && networkDataTask->isMainResource()) {
+            networkDataTask->networkLoadMetrics().interfaceName = String(metrics._interfaceName);
+            responseMetrics->interfaceName = networkDataTask->networkLoadMetrics().interfaceName;
```

Comment:
If there is a reason not to do something like the following, you should add a comment explaining it is necessary

```suggestion
            auto interfaceName = String(metrics._interfaceName);
            networkDataTask->networkLoadMetrics().interfaceName = interfaceName;
            responseMetrics->interfaceName = interfaceName;
```

Reply:
I removed this code, this was used to keep things consistent so that the metrics on the response and on end of load would be the same for interfaceName.